tests/cpp_unit_tests/test_current_sensor.cpp

```diff
@@ -33,125 +33,177 @@ TEST_CASE("Test current sensor") {
-                auto const sym_param = sym_current_sensor.calc_param<symmetric_t>();
-
-                CHECK(sym_param.angle_measurement_type == AngleMeasurementType::local_angle);
-                CHECK(sym_param.measurement.real_component.variance == doctest::Approx(pow(1.0 / base_current, 2)));
-                CHECK(sym_param.measurement.imag_component.variance == doctest::Approx(pow(0.2 / base_current, 2)));
-                CHECK(real(sym_param.measurement.value()) == doctest::Approx(1.0 / base_current));
-                CHECK(imag(sym_param.measurement.value()) == doctest::Approx(0.0 / base_current));
-            }
-
-            SUBCASE("90deg phase shift") {
-                sym_current_sensor.update(
-                    {.id = 1, .i_sigma = 1.0, .i_angle_sigma = 0.2, .i_measured = 1.0, .i_angle_measured = pi / 2});
-                auto const sym_param = sym_current_sensor.calc_param<symmetric_t>();
-
-                CHECK(sym_param.angle_measurement_type == AngleMeasurementType::local_angle);
-                CHECK(sym_param.measurement.real_component.variance == doctest::Approx(pow(0.2 / base_current, 2)));
-                CHECK(sym_param.measurement.imag_component.variance == doctest::Approx(pow(1.0 / base_current, 2)));
-                CHECK(real(sym_param.measurement.value()) == doctest::Approx(0.0 / base_current));
-                CHECK(imag(sym_param.measurement.value()) == doctest::Approx(1.0 / base_current));
-            }
-
-            SUBCASE("45deg phase shift") {
-                using std::numbers::sqrt2;
-                constexpr auto inv_sqrt2 = sqrt2 / 2;
-
-                sym_current_sensor.update(
-                    {.id = 1, .i_sigma = 1.0, .i_angle_sigma = 0.2, .i_measured = 1.0, .i_angle_measured = pi / 4});
-                auto const sym_param = sym_current_sensor.calc_param<symmetric_t>();
-
-                CHECK(sym_param.angle_measurement_type == AngleMeasurementType::local_angle);
-                CHECK(sym_param.measurement.real_component.variance ==
-                      doctest::Approx(1.04 / 2.0 / (base_current * base_current)));
-                CHECK(sym_param.measurement.imag_component.variance ==
-                      doctest::Approx(sym_param.measurement.real_component.variance));
-                CHECK(real(sym_param.measurement.value()) == doctest::Approx(inv_sqrt2 / base_current));
-                CHECK(imag(sym_param.measurement.value()) == doctest::Approx(real(sym_param.measurement.value())));
+            for (auto const terminal_type :
+                 {MeasuredTerminalType::branch_from, MeasuredTerminalType::branch_to, MeasuredTerminalType::branch3_1,
+                  MeasuredTerminalType::branch3_2, MeasuredTerminalType::branch3_3}) {
+                for (auto const angle_measurement_type :
+                     {AngleMeasurementType::global_angle, AngleMeasurementType::local_angle}) {
```

Comment:
here and elsewhere: you may want to define a constant for this to get rid of one nesting of loops. we already do something similar for the auto tap changer, e.g.

```cpp
namespace {
using TerminalAngleTypePair = std::pair<MeasuredTerminalType, AngleMeasurementType>;  // or std::tuple

constexpr auto terminal_types_angle_measurement_types = []{
    std::vector<TerminalAngleTypePair > result;  // or use an std::array
    for (auto const terminal_type :
         {MeasuredTerminalType::branch_from, MeasuredTerminalType::branch_to, MeasuredTerminalType::branch3_1,
          MeasuredTerminalType::branch3_2, MeasuredTerminalType::branch3_3}) {
        for (auto const angle_measurement_type :
             {AngleMeasurementType::global_angle, AngleMeasurementType::local_angle}) {
            result.emplace_back(terminal_type, angle_measuremen_type);
        }
    }
    return result;
}();
} // namespace
```
then here, you can do
```suggestion
            for (auto const [terminal_type, angle_measurement_type] : terminal_types_angle_measurement_types) {
```

Reply:
Nice. Appyling.